### ISSUE TITLE
docker lvm and filesystem issues

### ISSUE THREAD

**Issue by stevendborrelli** [CiscoCloud/microservices-infrastructure#1014]
The current implementation of mantl is not creating the docker filesystem correctly:

This is what I am seeing with overlay fs on a new system (with docker using `/var/lib/docker` instead of the mantl lv:

```
storage Driver: overlay
 Backing Filesystem: extfs
```
1. A docker lv should be in the `mantl`l lv and mounted at /var/lib/docker, not mounted as part of root.
2. The docker lv filesystem should be `xfs` instead of `extfs`.

For reference, here seems to be the most active docker issue: https://github.com/docker/docker/issues/10294

And RHEL's guidelines for only using xfs with overlay are at: https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/7.2_Release_Notes/technology-preview-file_systems.html

**Comment by stevendborrelli** [comment]
Just some things we need to do that docker-storage-setup doesn't:

1. create lv dockerlv: `vcreate -n dockerlv  -L 40G mantl`
2. create filesystem: `mkfs -t xfs /dev/mantl/dockerlv` 
3. update `/etc/fstab`: `/dev/mantl/dockerlv /var/lib/docker xfs defaults 0 0 `
4. mount /var/lib/docker
5. start docker

**Comment by stevendborrelli** [comment]
I like the way CoreOS does this via unit files (like docker-storage-setup). 
Just thinking about the different code paths:

extra devices available (aws, gce, openstack):
device mapper: create vg/lv (docker-storage-setup)
overlay: create vg/lv (docker storage-setup), xfs fs, /etc/fstab

```
#cloud-config
coreos:
  units:
    - name: format-ephemeral.service
      command: start
      content: |
        [Unit]
        Description=Formats the ephemeral drive
        After=dev-xvdb.device
        Requires=dev-xvdb.device
        [Service]
        Type=oneshot
        RemainAfterExit=yes
        ExecStart=/usr/sbin/wipefs -f /dev/xvdb
        ExecStart=/usr/sbin/mkfs.ext4 -F /dev/xvdb
    - name: var-lib-docker.mount
      command: start
      content: |
        [Unit]
        Description=Mount ephemeral to /var/lib/docker
        Requires=format-ephemeral.service
        After=format-ephemeral.service
        Before=docker.service
        [Mount]
        What=/dev/xvdb
        Where=/var/lib/docker
        Type=ext4
```

**Comment by ryane** [comment]
fixed by #1032
